File: leetcode75/rust/1431.rs
impl Solution {
    pub fn kids_with_candies (candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
       let &max_v = candies.iter().max().unwrap_or(&0);
       candies.into_iter().map(|c| c + extra_candies >= max_v).collect()
    }
}

/*
PSEUDOCODE:
FUNCTION kidsWithCandies(candies, extra):
    M := max(candies)
    result := empty list
    FOR each c IN candies:
        append (c + extra >= M) TO result
    RETURN result

Time: O(n). Space: O(n) for output.
*/

File: leetcode75/python/1431.py
from typing import List 

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int ) -> List[bool]:
        max_c = max(candies)
        return [c + extraCandies >= max_c for c in candies]
    


'''
FUNCTION kidsWithCandies(candies: array[int], extra: int) -> array[bool]
    M := max(candies)
    result := empty list
    FOR each c IN candies:
        append (c + extra >= M) TO result
    RETURN result

'''
